challenger/src/lib.rs

```diff
@@ -7,12 +7,18 @@ extern crate alloc;
 pub mod duplex_challenger;
 pub mod hash_challenger;
 
-use p3_field::field::Field;
+use p3_field::field::{Field, FieldExtension};
 
 /// Observes prover messages during an IOP, and generates Fiat-Shamir challenges in response.
 pub trait Challenger<F: Field> {
     fn observe_element(&mut self, element: F);
 
+    fn observe_ext_element<FE: FieldExtension<F>>(&mut self, ext: FE) {
+        for coeff in ext.as_base_slice() {
+            self.observe_element(*coeff);
+        }
+    }
```

Comment:
Rather minor, but I wonder if this wouldn't be better as
```suggestion
    fn observe_elements(&mut self, elts: &[F]) {
        for &elt in elts {
            self.observe_element(elt);
        }
    }
```
([same as here](https://github.com/mir-protocol/plonky2/blob/f11921c9ad672bb2cae28253db25bacc06bb2655/plonky2/src/iop/challenger.rs#L65)) and then require that callers who have a `FieldExtension` call `challenger.observe_elements(fe.as_base_slice())`? The main advantage is just to remove the dependence on `FieldExtension` (which I'm still inclined to remove altogether). Anyway, your call in this case.

Reply:
I think we'd end up with similar code if we remove `FieldExtension`? I.e. `<FE: Field<Base = F>>` vs `<FE: FieldExtension<F>>`.

We could start with `observe_elements` (and it's definitely something we should have), but in Plonky2 `get_extension_challenge` was used quite a bit, and the inlined version of that seems a bit more verbose. Plonky2 also had a couple other extension methods which I think may be handy later.

Reply:
What if `as_base_slice()` was just an element of `Field`?

Main motivation more generally is to think of ways to avoid having functions that exist in two versions, one for `Field` and one for `FieldExtension`; it's obviously a pain to maintain, and often conceptually untidy. Personally, I'm convinced that it should rarely, if ever, be necessary to treat `Field` and `FieldExtension` separately; but maybe this particular case is an exception?